### PR TITLE
Refactor Questions Entry - LRQA-82286 | master

### PR DESCRIPTION
**Pull request by Italo-gabriel**
**JIRA** [LRQA-82286](https://liferay.atlassian.net/browse/LRQA-82286)

### CODE REVIEW

**Comment by Italo-gabriel** [comment]
ci:test:sf

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/collaboration/questions/QuestionsEntry.testcase [hunk 1]
```diff
@@ -40,79 +40,93 @@ definition {
...
-
-		JSONDepot.connectSite(
-			depotName = "Test Depot Name",
-			groupName = "Guest");
-
-		JSONMBMessage.addCategory(
-			categoryDescription = "Description",
-			categoryName = "Questions Test Category",
-			userEmailAddress = "test@liferay.com");
+		task ("Given a depot image is Adding to a question and connected a site") {
```
Comment (nikki-pru) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/collaboration/questions/QuestionsEntry.testcase:47]: Given a connected depot with an image

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/collaboration/questions/QuestionsEntry.testcase [hunk 2]
```diff
@@ -40,79 +40,93 @@ definition {
 		}
 	}
 
-	@description = "Validate if the user can create a question with an AL image."
+	@description = "LPS-110392 - Verifies if the user can create a question with an AL image."
```
Comment (nikki-pru) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/collaboration/questions/QuestionsEntry.testcase:43]: LPS-110392 - Verify if the user can create a question with an image from a connected Depot

Comment (nikki-pru) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/collaboration/questions/QuestionsEntry.testcase:43]: LPS-110392 - Verify if the user can create a question with an image from a connected Depot

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/collaboration/questions/QuestionsEntry.testcase [hunk 3]
```diff
@@ -40,79 +40,93 @@ definition {
...
 
-		Questions.askQuestionFillingTheMandatoryFields(
-			contentField = "Body AddQuestionWithTheALImage",
-			depotName = "Test Depot Name",
-			imageFileName = "Document_2.jpeg",
-			layoutName = "questions-page",
-			siteURLKey = "guest",
-			titleField = "Title AddQuestionWithTheALImage",
-			topicName = "questions-test-category");
+		task ("And has added a category") {
```
Comment (nikki-pru) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/collaboration/questions/QuestionsEntry.testcase:64]: and a Message Boards Category is added

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/collaboration/questions/QuestionsEntry.testcase [hunk 4]
```diff
@@ -40,79 +40,93 @@ definition {
...
+			JSONMBMessage.addCategory(
+				categoryDescription = "Description",
+				categoryName = "Questions Test Category",
+				userEmailAddress = "test@liferay.com");
+		}
 
-		Questions.gotoDetails(
-			questionTabField = "Questions",
-			questionTitle = "Title AddQuestionWithTheALImage");
+		task ("When Questions ask a question filling the mandatory fields") {
```
Comment (nikki-pru) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/collaboration/questions/QuestionsEntry.testcase:71]: When a Question is created with an image from the connected Depot

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/collaboration/questions/QuestionsEntry.testcase [hunk 5]
```diff
@@ -40,79 +40,93 @@ definition {
...
+				titleField = "Title AddQuestionWithTheALImage",
+				topicName = "questions-test-category");
+
+			Questions.gotoDetails(
+				questionTabField = "Questions",
+				questionTitle = "Title AddQuestionWithTheALImage");
+		}
 
-		Questions.viewImageInQuestion(imageFileName = "document_2-jpeg");
+		task ("Then The image associated with the question should be displayed.") {
```
Comment (nikki-pru) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/collaboration/questions/QuestionsEntry.testcase:86]: Then the image from the Depot should be displayed

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/collaboration/questions/QuestionsEntry.testcase [hunk 6]
```diff
@@ -40,79 +40,93 @@ definition {
...
+			Questions.viewImageInQuestion(imageFileName = "document_2-jpeg");
+		}
 	}
 
-	@description = "Validate if the user can create a question with an image."
+	@description = "LPS-110392 - Verifies if the user can create a question with an image."
```
Comment (nikki-pru) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/collaboration/questions/QuestionsEntry.testcase:91]: LPS-110392 - Verify if the user can create a question with an image from the DM Library

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/collaboration/questions/QuestionsEntry.testcase [hunk 7]
```diff
@@ -40,79 +40,93 @@ definition {
...
 	test CanAddQuestionWithImage {
 		property portal.acceptance = "true";
 
-		JSONDocument.addFileWithUploadedFile(
-			dmDocumentDescription = "DM Document Description",
-			dmDocumentTitle = "Document_1.svg",
-			groupName = "Guest",
-			mimeType = "image/svg+xml",
-			sourceFileName = "Document_1.svg");
+		task ("Given an image and a category is adding to a question") {
```
Comment (nikki-pru) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/collaboration/questions/QuestionsEntry.testcase:98]: Given an image exists in Document Library and a category is available

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/collaboration/questions/QuestionsEntry.testcase [hunk 8]
```diff
@@ -40,79 +40,93 @@ definition {
...
 
-		Questions.gotoCategory(categoryName = "Questions Test Category");
+			Questions.gotoCategory(categoryName = "Questions Test Category");
+		}
 
-		Questions.askQuestionFillingTheMandatoryFields(
-			contentField = "Body AddQuestionWithTheImage",
-			imageFileName = "Document_1.svg",
-			titleField = "Title AddQuestionWithTheImage");
+		task ("When Questions ask a question filling the mandatory fields") {
```
Comment (nikki-pru) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/collaboration/questions/QuestionsEntry.testcase:116]: When a Question is created with an image from Document Library

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/collaboration/questions/QuestionsEntry.testcase [hunk 9]
```diff
@@ -40,79 +40,93 @@ definition {
...
-		Questions.gotoDetails(
-			questionTabField = "Questions",
-			questionTitle = "Title AddQuestionWithTheImage");
+			Questions.gotoDetails(
+				questionTabField = "Questions",
+				questionTitle = "Title AddQuestionWithTheImage");
+		}
 
-		Questions.viewImageInQuestion(imageFileName = "document_1-svg");
+		task ("Then The image associated with the question should be displayed.") {
```
Comment (nikki-pru) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/collaboration/questions/QuestionsEntry.testcase:127]: Then the image from Document Library should be displayed

**Review by nikki-pru — CHANGES_REQUESTED** [review]
Minor changes for the task blocks to differentiate that one test is using an image from a connected depot and the other test is using an image from Document Library

**Comment by nikki-pru** [comment]
ci:test:sf

**Comment by nikki-pru** [comment]
ci:test:questions

**Review by nikki-pru — APPROVED** [review]

**Comment by nikki-pru** [comment]
ci:stop

**Comment by nikki-pru** [comment]
Failures are unrelated. Forwarding.

**Comment by nikki-pru** [comment]
ci:test:forward

**Comment by MGabriellaS** [comment]
ci:forward

**Pull request closed without merging**